### DISCUSSION TITLE
@db.Time and @db.Date is not working

### DISCUSSION THREAD

**Discussion by vivekjpatel** [prisma/prisma#21568] [Q&A] [answered]
### Question

Hi Everyone!

I am using MsSQL Server Database. I have model as per below.
```prisma
model sample {
  startTime      DateTime          @db.Time
}
```

And versions are as per below.
"@prisma/client": "^5.4.2",
"prisma": "^5.4.2",

But whenever I insert data with format `"startTime": "15:30:00",`. It's not working and prisma throws error like **"Invalid value for argument `startTime`: input contains invalid characters. Expected ISO-8601 DateTime**."

Can you please help on this ?

Thanks

### Expected behavior (optional)

It should store the time as we are providing.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @vivekjpatel 👋 

Thank you for raising this question.

You are getting this error because Prisma expects DateTime values to be in [ISO-8601](https://www.iso.org/iso-8601-date-and-time-format.html) format. However, you're trying to insert a time value in the format "HH:MM:SS", which is not compatible with the ISO-8601 DateTime format.

To resolve this issue, you could try to include a date part with your time value. The date part could be a fixed value, like "1970-01-01". So your time value would look something like this: "1970-01-01T15:30:00Z".
Here's how you could modify your insert operation:
```js
prisma.sample.create({
  data: {
    startTime: new Date("1970-01-01T15:30:00Z"),
  },
});
```
This will create a JavaScript Date object, which is in the `ISO-8601` format that Prisma expects for DateTime values. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by vivekjpatel** [reply]
Hi @ludralph,

Thanks for your time and answer. So Prisma expects ISO-8601 format only right? It don't care about datatype provide for database.

Thanks

**Reply by figassis** [reply]
HH:MM:SS is valid ISO-8601 https://fits.gsfc.nasa.gov/iso-time.html

**Reply by scalprox** [reply]
> HH:MM:SS is valid ISO-8601 https://fits.gsfc.nasa.gov/iso-time.html

Prisma expect ISO-8601 DateTime. But HH:MM:SS is only a Time
